### ISSUE TITLE
与mi-aqara冲突，超时问题

### ISSUE THREAD

**Issue by Raccooon** [LASER-Yi/homebridge-mi-acpartner#40]
单独运行ac-parterner和mi-aqara都没有问题
但是放在一个桥里面，mi-aqara会超时
貌似mi-aqara的子设备没加载完，空调就是开始加载了
导致其一直读不到返回值？
```
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd": "whois"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [INFO]server is listening on port 9898.
[7/13/2018, 11:33:03 AM] Homebridge is running on port 51826.
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"iam","port":"9898","sid":"7811dcdc2a79","model":"gateway","ip":"192.168.1.132"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"get_id_list"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"get_id_list_ack","sid":"7811dcdc2a79","token":"psGMUgXMweX8tj3S","data":"[\"158d0001e77bcd\",\"158d0001a1fc1e\",\"158d00016146c9\"]"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"7811dcdc2a79"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"read_ack","model":"gateway","sid":"7811dcdc2a79","short_id":0,"data":"{\"rgb\":0,\"illumination\":672,\"proto_version\":\"1.0.9\"}"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:03 AM] [MiAqaraPlatform] [DEBUG]read gateway device list finished. size: 3
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] [INFO]Device 192.168.1.250 -> Connected
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] [Ac Partner]Ready
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] [WARN]Sync off
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] Sync -----------------------------
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] Accessory -> Ac Partner
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] Model -> 0100011272
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] Active -> 0
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] Mode -> 1
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] Temperature -> 26
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] RotationSpeed -> 3
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] SwingMode -> 1
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] LED -> 0
[7/13/2018, 11:33:04 AM] [XiaoMiAcPartner] -----------------------------------
[7/13/2018, 11:33:06 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:06 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:06 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:09 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:09 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:09 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:12 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:12 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:12 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:15 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:15 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:15 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:18 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:18 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:18 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:21 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:21 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:21 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:24 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:24 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:24 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:27 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:27 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:27 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:30 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:30 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:30 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:33 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:33 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:33 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:36 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:36 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:36 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:39 AM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 11:33:39 AM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"read", "sid":"158d0001e77bcd"}
    at Timeout.setInterval (/usr/lib/node_modules/homebridge-mi-aqara/index.js:399:27)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
[7/13/2018, 11:33:39 AM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 11:33:39 AM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"read", "sid":"158d0001a1fc1e"}
    at Timeout.setInterval (/usr/lib/node_modules/homebridge-mi-aqara/index.js:399:27)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
[7/13/2018, 11:33:39 AM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"read", "sid":"158d00016146c9"}
[7/13/2018, 11:33:39 AM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"read", "sid":"158d00016146c9"}
    at Timeout.setInterval (/usr/lib/node_modules/homebridge-mi-aqara/index.js:399:27)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```

**Comment by Raccooon** [comment]
贴一下config
`{
        "bridge": {
                "name": "Homebridge",
                "username": "F1:35:ED:EF:AC:26",
                "port": 51826,
                "pin": "233-26-913"
        },
        "platforms": [{
                        "platform": "MiAqaraPlatform",
                        "gateways": {
                                "7811dcdc2a79": "460521AE11F04270"
                        },
                        "defaultValue": {
                                "7811dcdc2a79": {
                                        "Gateway_Lightbulb": {
                                                "name": "夜灯"
                                        },
                                        "Gateway_LightSensor": {
                                                "name": "光线传感器"
                                        },
                                        "Gateway_Switch_JoinPermission": {
                                                "name": "加入树洞吧",
                                                "disable": true
                                        }
                                },
                                "158d0001e77bcd": {
                                        "ContactSensor_ContactSensor": {
                                                "name": "窗户"
                                        }
                                },
                                "158d0001a1fc1e": {
                                        "TemperatureAndHumiditySensor2_TemperatureSensor": {
                                                "name": "温度传感器"
                                        },
                                        "TemperatureAndHumiditySensor2_HumiditySensor": {
                                                "name": "湿度传感器"
                                        }
                                },
                                "158d00016146c9": {
                                        "SingleSwitch_Switch": {
                                                "name": "灯",
                                                "serviceType": "Lightbulb"
                                        }
                                }
                        }
                },
                {
                        "platform": "XiaoMiAcPartner",
                        "devices": {
                                "192.168.1.250": "a63258eb6a86d85939791dcca26f3917"
                        },
                        "accessories": [{
"name": "Ac Partner",
                                "type": "climate",
                                "customize": {
                                        "off": "010001127201311a01004DB2DE2107F8",
                                        "on": "",
                                        "auto": "",
                                        "heat": {
                                                "30": "",
                                                "29": "",
                                                "17": ""
                                        },
                                        "cool": {
                                                "30": "010001127211311e11004DB2FD020DF2",
                                                "29": "010001127211311d11004DB2FD0205FA",
                                                "28": "010001127211311c11004DB2FD0201FE",
                                                "27": "010001127211311b11004DB2FD0209F6",
                                                "26": "010001127211311a11004DB2FD020BF4",
                                                "25": "010001127211311911004DB2FD0203FC",
                                                "24": "010001127211311811004DB2FD0202FD",
                                                "23": "010001127211311711004DB2FD020AF5",
                                                "22": "010001127211311611004DB2FD020EF1",
                                                "21": "010001127211311511004DB2FD0206F9",
                                                "20": "010001127211311411004DB2FD0204FB",
                                                "19": "010001127211311311004DB2FD020CF3",
                                                "18": "010001127211311211004DB2FD0208F7",
                                                "17": "010001127211311111004DB2FD0200FF2"
                                        }
                                }
                        }]
                }
        ]
}
`

**Comment by Raccooon** [comment]
`
[7/13/2018, 8:53:56 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001e77bcd"}
[7/13/2018, 8:53:56 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d0001a1fc1e"}
[7/13/2018, 8:53:56 AM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"read", "sid":"158d00016146c9"}
`
插件会一直尝试加载我之前设定好的子设备，不一会就超时

**Comment by Raccooon** [comment]
感谢您的付出，给我们开发了这么好的插件
折腾了一通宵，暂时剩这两个问题没解决
恳请能得到您的反馈

**Comment by hlouis** [comment]
是的，最近也遇到这个问题，mi-aqara 和 ac partner 跑在同一个 docker 的 bridge 里面之后，aqara 中的所有设备的操作都会超时，但是状态还是可以读取。

在 config 中移除了 XiaoMiAcPartner 整个 platform 之后，重启 docker ，之前的问题就都消失了，麻烦看看是什么原因。

**Comment by h605290713** [comment]
我也是 两个在一起就不能使用了

**Comment by shunnokw** [comment]
Same issue. Please help

**Comment by LASER-Yi** [comment]
这个问题主要是因为miio版本更新了，我之前有向miio的作者反馈过，看来他还没有修改，会在0.7.1降回到原来的miio版本

**Comment by LASER-Yi** [comment]
Fixed, please upgrade to v0.7.1

**Comment by Raccooon** [comment]
```
[7/24/2018, 12:58:26 PM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"report","model":"ctrl_neutral1","sid":"158d00016146c9","short_id":7781,"data":"{\"channel_0\":\"on\"}"}
[7/24/2018, 12:58:26 PM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"on\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:26 PM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"report","model":"ctrl_neutral1","sid":"158d00016146c9","short_id":7781,"data":"{\"channel_0\":\"on\"}"}
[7/24/2018, 12:58:27 PM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"on\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:28 PM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"on\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:28 PM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"on\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
    at Timeout.setInterval (/usr/lib/node_modules/homebridge-mi-aqara/index.js:399:27)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
[7/24/2018, 12:58:32 PM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"off\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:33 PM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"report","model":"ctrl_neutral1","sid":"158d00016146c9","short_id":7781,"data":"{\"channel_0\":\"off\"}"}
[7/24/2018, 12:58:33 PM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"off\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:34 PM] [MiAqaraPlatform] [DEBUG][Revc]{"cmd":"report","model":"ctrl_neutral1","sid":"158d00016146c9","short_id":7781,"data":"{\"channel_0\":\"off\"}"}
[7/24/2018, 12:58:34 PM] [MiAqaraPlatform] [DEBUG][Send]{"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"off\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:35 PM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"off\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
[7/24/2018, 12:58:35 PM] [MiAqaraPlatform] [ERROR]Error: timeout: {"cmd":"write","model":"ctrl_neutral1","sid":"158d00016146c9","data":"{\"channel_0\":\"off\", \"key\": \"7e2bd4b3324484429629d6782f52cfd4\"}"}
    at Timeout.setInterval (/usr/lib/node_modules/homebridge-mi-aqara/index.js:399:27)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```
能用是能用了，但是貌似还会报超时错误
